### ISSUE TITLE
use:links doesn't work when target attribute is set on anchor

### ISSUE THREAD

**Issue by nj-vs-vh** [EmilTholin/svelte-routing#260]
Hi!

This is probably a general issue, but I've encountered it specifically debugging why `use:links` directive doesn't work with [SvelteUI](https://www.svelteui.org/core/button)'s buttons. Turns out [they always set](https://github.com/svelteuidev/svelteui/blob/e2b4e5e2d95c616224b15dfe8107efd863beb636/packages/svelteui-core/src/components/Button/Button.svelte#L96) `target` attribute to non-`''` value and `links` expects it to not be set.

Probably fixable by changing [this line](https://github.com/EmilTholin/svelte-routing/blob/4582a86a0c8067c5599a5ca7809933d0012ae9cd/src/actions.js#L66) to 
```javascript
anchor.target !== "_blank" &&
```
or at least to
```javascript
(anchor.target === "" || anchor.target === "_self") &&
```

`_self` is [the default value](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a), but I think it would be good to support the case when it is explicitly set and not just omitted

**Comment by krishnaTORQUE** [comment]
Is your issue resolved?

**Comment by nj-vs-vh** [comment]
I have found the workaround, but the issue in the library code is still there. Sorry for the hectic description, here is the minimal reproducible example:

```svelte
<script lang="ts">
  import { Router, Route, links } from "svelte-routing";
</script>

<div use:links>
  <Router>
    <Route path="/one">
      Path one
      <br />
      <a href="/two" target="_self">Go to two</a>
    </Route>
    <Route path="/two">Path two</Route>
  </Router>
</div>
```

In the example above click on `<a href="/two" target="_self">Go to two</a>` should be catched by the library (`use:links`), but it's not. So it causes the full page reload, elimitaning all advantages of client-side routing.

As mentioned above, this is caused by `target="_self"` on the anchor.

**Comment by nj-vs-vh** [comment]
I have found out about the bug by trying to use it with SvelteUI, yes. But `target="_self"` is a perfectly valid attribute on an anchor, as per [specification](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a). So SvelteUI is irrelevant, really. Sorry for the confusion, I should have been more clear from the start. I have provided the MRE without any additional libraries.

**Comment by krishnaTORQUE** [comment]
This issue has been fixed in v2.5.0
